packages/compile-solidity/src/profiler/loadParser.ts

```diff
@@ -56,12 +63,14 @@ function makeParseImports(parser) {
         // if there's a match provide the helpful error, otherwise return solc's error output
         if (contractSolcPragma) {
           const contractSolcVer = contractSolcPragma[0];
-          const configSolcVer = semver.valid(solc.version());
+          const configSolcVer = semver.valid(parser.version());
```

Comment:
Logic change. Are we sure that we want `parser.version` instead of `solc.version` here? Can they ever be different versions?

Reply:
This was an error. There is no `solc` in this scope and rather `parser` is the variable we want to use. If you look at the call to this method ~ line 24 you can see `solc` is passed in here (a portion of the result of calling `supplier.load`). Looks to me like someone got their names confused.

Reply:
That begs the question as to whether this code is even reachable... Not something to worry about in this PR if we're aiming for a homomorphic change here, however.

Reply:
Do you think I'm mistaken? I think this code will crash if executed. I would argue that this is a small enough change that it shouldn't be a big deal to include it. What do you think about that point of view?

Reply:
> Do you think I'm mistaken?

Oh not at all! My reply was more academic, sorry. It's more that if the original code would crash if executed, but we weren't ever seeing that, this whole path may be effectively dead code.

I don't think there's any further action required here. Sorry for the confusion!